internal/pkg/agent/cmd/enroll.go

```diff
@@ -473,7 +481,8 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command) error {
 	key, _ := cmd.Flags().GetString("elastic-agent-cert-key")
 	keyPassphrase, _ := cmd.Flags().GetString("elastic-agent-cert-key-passphrase")
 
-	ctx := handleSignal(context.Background())
+	ctx, cancel := context.WithTimeout(handleSignal(context.Background()), enrollTimeout)
```

Comment:
What happens here if `--enroll-timeout 0` or `--enroll-timeout -1` is passed? Does `0` result in this completing instantly? I think `-1` would make it go to never timing out, I don't think that is what would happen here.

Reply:
I think a negative value would time out instantly, but I make it so a zero/negative value disables the timeout